src/controllers/queries/index.ts

```diff
@@ -500,6 +508,81 @@ export class QueriesController extends HTMLController {
       })
     )
   }
+
+  private async submitDrpcRequest({
```

Comment:
Ok I think there's some confusion here as to what's going on. As far as I understand this method:

1. inserts a new `query` into the `db` and generates a new `id` for that query
2. Updates the original `query` with our local contribution.
3. Sends the `drpc` request to `charlie` using the newly inserted row
4. Adds `query_rpc` row afterwards to track that this has happened

So there's a few issues:

1. This is happening in a loop so step 2 is happening multiple times. I'd pull that out of this method
2. That should also allow you to deduplicate this functionality in `scope3CarbonConsumptionStage`
3. We need a way of linking the new query with the parent. This will require new structure in the `query` table. Maybe a `nullable` column `parent_id` with a foreign key back to the same table

Please let me know if any of this is wrong

Reply:
this has been updated along with new column